### ISSUE TITLE
mAP score not improving with multi gpu training

### ISSUE THREAD

**Issue by sanchit2843** [ultralytics/ultralytics#884]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,
I am training yolov8m with an image size of 640. I tried the training models on two machines, 8 A100 and 1 3090, with batch sizes 512 and 16, respectively.
My mAP score stays between 0.28-0.31 for over 60 epochs in the case of 8 A100. On the other hand, my mAP score after just 1 epoch with 1 3090 is 0.591. I am unable to understand the issue; the yolov5 works well in similar scenarios. Is there any specific hyperparameter or argument I need to tune? 
Looking forward to hearing back.
Regards

**Comment by Laughing-q** [comment]
@sanchit2843 that sounds strange...actually some of our coco models are trained by single-gpu and some of them are trained by multi-gpu, all of the coco models look good to me. can you provide your training command?

**Comment by sanchit2843** [comment]
Hi, I tried training with a single A100 for 10-15 epochs and the model didn't improve, on the other hand I start getting 0.59 mAP on 3090 after just 1 epoch. The training command I used is same on both the systems. 

Here is the command for single GPU training:

`yolo task=detect mode=train model=yolov8m.pt data=/home/ubuntu/detection/data/coco.yaml epochs=100 imgsz=640 batch=16 
`
Here is the command for multiple GPU

yolo task=detect mode=train model=yolov8m.pt data=/home/ubuntu/detection/data/coco.yaml epochs=100 imgsz=640  device=\'0,1,2,3,4,5,6,7\'

**Comment by pderrenger** [comment]
@sanchit2843 the issue may be related to the batch size and its impact on training. When training with smaller batch sizes (e.g., 16 on a single 3090 GPU), the mAP improves significantly after just 1 epoch, while training with larger batch sizes (e.g., 512 on 8 A100 GPUs) results in slower mAP improvement.

Reducing the batch size for multi-GPU training could potentially lead to better mAP performance. Additionally, the learning rate might need adjustment when altering the batch size.

Try reducing the batch size for multi-GPU training and adjusting the learning rate accordingly to see if it positively impacts the mAP score.
